### ISSUE TITLE
[Enhancement] suppress prepare-commit-msg temporarily / permanent

### ISSUE THREAD

**Issue by LW-archlinux** [eli-schwartz/aurpublish#11]
I do like the prepare-commit-msg hook, but feel it's overkill for packages that change package versions frequently like VCS packages.

An option to suppress the hook temporarily or permanently for specific packages would help .

Possible invocations :
```aurpublish -rm_auto_commit_hook PACKAGE```
```aurpublish -suppress_auto_commit_hook_once PACKAGE```

**Comment by eli-schwartz** [comment]
I don't often update VCS packages -- only when there are upstream changes to the build system.

Incidentally, what do you see as overkill about it? Note that it's used at `git commit` time, so I'm not sure what it would look like to disable with options to aurpublish anyway.

**Comment by LW-archlinux** [comment]
I often add patches to solve something and remove them after they've been accepted to master. Also llvm trunk changes build options rather often .

The overkill happens mainly because I try to keep commits small to make it easier to test changes.
An example are the last 5 changes at https://aur.archlinux.org/cgit/aur.git/log/?h=llvm-git . They happened within a few hours and I only pushed them after the last commit.

A hook that is triggered by git push instead of git commit could solve that, is such a hook possible ?

**Comment by eli-schwartz** [comment]
> A hook that is triggered by git push instead of git commit could solve that, is such a hook possible ?

No, the hook prepares a commit message and must therefore be run at the time you create the commit... on push can't affect that.

One possibility might be to have a config file in the repository which the hook would read, and use to disable its commit-msg suggestion for certain packages (or alternatively, to provide a different one).

**Comment by eli-schwartz** [comment]
Is this something you're interested in investigating and possibly providing a patch to implement it?
